app/src/main/java/com/xmartlabs/template/controller/ServiceAndDatabaseController.java

```diff
@@ -0,0 +1,140 @@
+package com.xmartlabs.template.controller;
+
+import android.support.annotation.CheckResult;
+import android.support.annotation.NonNull;
+
+import com.raizlabs.android.dbflow.sql.language.property.Property;
+import com.xmartlabs.template.helper.function.BiFunction;
+import com.xmartlabs.template.helper.function.Function;
+import com.xmartlabs.template.model.EntityWithId;
+
+import java.util.List;
+
+import io.reactivex.Completable;
+import io.reactivex.Flowable;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+@SuppressWarnings({"unused", "WeakerAccess"})
+public abstract class ServiceAndDatabaseController<Id, D extends EntityWithId<Id>, C> extends Controller {
```

Comment:
there are a lot of protected functions that looks reasonable to have as public. Some examples: `createEntity ` `deleteEntity`. Why was that decision? Doesn't having them accessible from other classes make sense?